### ISSUE TITLE
Should stop using the /device-types/v1 in favor of the hostApp model in takeover

### ISSUE THREAD

**Issue by thgreasi** [balena-os/takeover#18]
Example from the image maker
See: https://github.com/balena-io/balena-img/pull/337/files#diff-769911c416ccf8514d8fd941ae0abe8fb5c606ade0c218e22151a5f5f9f3d700R269

See: https://cs.github.com/balena-os/takeover/blob/43c6874ea2f918d3bfd6cbb6f985da31574406ca/src/stage1/api_calls.rs#L30
See: https://jel.ly.fish/improvement-stop-relying-device-types-v1-device-type-json-unrelated-things-4fbac3c

**Comment by thgreasi** [comment]
Example curls that do not rely on release_tags since we are going to drop those soon:
```
-- curl that gets the latest available (succesful, non draft, non invalidated) non-ESR OS version for a given DT slug, ~ like the UI does
curl 'https://api.balena-cloud.com/v6/release?$top=1&$select=raw_version&$filter=(is_final%20eq%20true)%20and%20(is_passing_tests%20eq%20true)%20and%20(is_invalidated%20eq%20false)%20and%20(status%20eq%20%27success%27)%20and%20(belongs_to__application/any(bta:((bta/is_host%20eq%20true)%20and%20(bta/is_for__device_type/any(iodt:iodt/slug%20eq%20%27raspberrypi3%27)))%20and%20(not(bta/application_tag/any(at:at/tag_key%20eq%20%27release-policy%27))%20or%20(bta/application_tag/any(at:(at/tag_key%20eq%20%27release-policy%27)%20and%20(at/value%20eq%20%27default%27))))))&$orderby=created_at%20desc' \
  -H 'authorization: Bearer <BEARER>'

-- curl that gets all available OS versions, including ESR, for a given DT slug, ~ like the UI does
curl 'https://api.balena-cloud.com/v6/release?$select=raw_version&$filter=(is_final%20eq%20true)%20and%20(is_passing_tests%20eq%20true)%20and%20(is_invalidated%20eq%20false)%20and%20(status%20eq%20%27success%27)%20and%20(belongs_to__application/any(bta:(bta/is_host%20eq%20true)%20and%20(bta/is_for__device_type/any(iodt:iodt/slug%20eq%20%27raspberrypi3%27))))&$orderby=created_at%20desc' \
  -H 'authorization: Bearer <BEARER>'

-- curl that checks whether a given OS version is available for a given DT slug
curl 'https://api.balena-cloud.com/v6/release?$select=raw_version&$filter=(raw_version%20eq%20%275.1.20%2Brev1%27)%20and%20(is_final%20eq%20true)%20and%20(is_passing_tests%20eq%20true)%20and%20(is_invalidated%20eq%20false)%20and%20(status%20eq%20%27success%27)%20and%20(belongs_to__application/any(bta:(bta/is_host%20eq%20true)%20and%20(bta/is_for__device_type/any(iodt:iodt/slug%20eq%20%27raspberrypi3%27))))' \
  -H 'authorization: Bearer <BEARER>'
```

**Comment by rahul-thakoor** [comment]
closed with #55
